File: ender-does-backend/src/main/java/com/enderdev/enderdoesbackend/todo/mappers/TodoMapper.java
package com.enderdev.enderdoesbackend.todo.mappers;

import com.enderdev.enderdoesbackend.todo.dto.TodoResponse;
import com.enderdev.enderdoesbackend.todo.models.Todo;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;

@Service
@RequiredArgsConstructor
public class TodoMapper {
    public TodoResponse toTodoResponse(Todo todo){
        return TodoResponse.builder()
                .id(todo.getId())
                .title(todo.getTitle())
                .ownerId(todo.getOwner().getId())
                .body(todo.getBody())
                .isDone(todo.getIsDone())
                .completedAt(todo.getCompletedAt() != null ? todo.getCompletedAt().toString() : "")
                .createdAt(todo.getCreatedAt().toString())
                .build();
    }
}

File: ender-does-backend-NET/Todo/Mappers/TodoMapper.cs
using ender_does_backend_NET.Todo.DTOs;
using ender_does_backend_NET.Todo.Models;

namespace ender_does_backend_NET.Todo.Mappers;

public static class TodoMapper
{
    public static TodoResponse ToResponse(Models.Todo todo)
    {
        return new TodoResponse(
            todo.Id,
            todo.Title,
            todo.Body,
            todo.IsDone,
            todo.CreatedAt,
            todo.CompletedAt
        );
    }
}